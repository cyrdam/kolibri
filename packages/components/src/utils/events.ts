// TODO: Should be synchronized with enums/events.ts
export type KoliBriEventType =
	| 'blur'
	| 'change'
	| 'changePage'
	| 'changePageSize'
	| 'click'
	| 'close'
	| 'focus'
	| 'input'
	| 'mousedown'
	| 'reset'
	| 'selection-change' // todo remove
	| 'selectionChange'
	| 'sort'
	| 'submit'
	| 'toggle';

export function stopPropagation(event: Event): void {
	event.stopImmediatePropagation();
	event.stopPropagation();
}

function createKoliBriEvent<T>(type: KoliBriEventType, detail: T): CustomEvent {
	const event = new CustomEvent(`kol-${type}`, {
		bubbles: true,
		cancelable: true,
		composed: true,
		detail: detail,
	});
	return event;
}

function dispatchKoliBriEvent<T>(target: EventTarget, type: KoliBriEventType, detail?: T): boolean {
	const dispatch = target.dispatchEvent(createKoliBriEvent(type, detail));
	return dispatch;
}

/**
 * @deprecated
 */
export function tryToDispatchKoliBriEvent<T>(type: KoliBriEventType, target?: EventTarget, detail?: T): void {
	target && dispatchKoliBriEvent(target, type, detail);
}

export function dispatchDomEvent(target: HTMLElement, type: KoliBriEventType) {
	target.dispatchEvent(new Event(type, { bubbles: true, composed: true }));
}

export function preventDefaultAndStopPropagation(event: Event) {
	event.preventDefault();
	stopPropagation(event);
}
