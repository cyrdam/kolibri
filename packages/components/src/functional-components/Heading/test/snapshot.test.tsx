import { h } from '@stencil/core';
import KolHeadingFc from '../Heading';
import { renderFunctionalComponentToSpecPage } from '../../../utils/testing';

describe('KolHeadingFc', () => {
	it('should render with default props', async () => {
		const page = await renderFunctionalComponentToSpecPage(() => <KolHeadingFc>Default Heading</KolHeadingFc>);

		expect(page.root).toBeDefined();
		expect(page.root).toMatchSnapshot();
		expect(page.root?.firstElementChild?.tagName).toBe('H1');
		expect(page.root?.textContent).toContain('Default Heading');
		expect(page.root?.classList.contains('kol-heading')).toBe(true);
	});

	it('should render with a specific level', async () => {
		const page = await renderFunctionalComponentToSpecPage(() => <KolHeadingFc level={3}>Level 3 Heading</KolHeadingFc>);

		expect(page.root).toBeDefined();
		expect(page.root).toMatchSnapshot();
		expect(page.root?.firstElementChild?.tagName).toBe('H3');
		expect(page.root?.textContent).toContain('Level 3 Heading');
		expect(page.root?.classList.contains('kol-heading')).toBe(true);
	});

	it('should render with a secondary headline', async () => {
		const page = await renderFunctionalComponentToSpecPage(() => <KolHeadingFc secondaryHeadline="Secondary Headline">Main Heading</KolHeadingFc>);

		expect(page.root).toBeDefined();
		expect(page.root).toMatchSnapshot();

		expect(page.root?.tagName).toBe('HGROUP');
		expect(page.root?.querySelector('h1')).not.toBeNull();
		expect(page.root?.querySelector('h2')).not.toBeNull();
		expect(page.root?.textContent).toContain('Main Heading');
		expect(page.root?.textContent).toContain('Secondary Headline');
		expect(page.root?.classList.contains('kol-heading')).toBe(true);
		expect(page.root?.querySelector('h1')?.classList.contains('kol-heading')).toBe(false);
		expect(page.root?.querySelector('h2')?.classList.contains('kol-heading')).toBe(false);
	});

	it('should apply custom class names', async () => {
		const page = await renderFunctionalComponentToSpecPage(() => <KolHeadingFc class="custom-class">Custom Class Heading</KolHeadingFc>);

		expect(page.root).toBeDefined();
		expect(page.root).toMatchSnapshot();
		expect(page.root?.firstElementChild?.tagName).toBe('H1');
		expect(page.root?.firstElementChild?.classList.contains('custom-class')).toBe(true);
		expect(page.root?.classList.contains('kol-heading')).toBe(true);
		expect(page.root?.textContent).toContain('Custom Class Heading');
	});
});
