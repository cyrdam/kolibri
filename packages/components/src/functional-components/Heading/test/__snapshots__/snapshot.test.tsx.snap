// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KolHeadingFc should apply custom class names 1`] = `
<h1 class="custom-class kol-heading--h1 kol-heading__headline">
  Custom Class Heading
</h1>
`;

exports[`KolHeadingFc should render with a secondary headline 1`] = `
<hgroup>
  <h1 class="kol-heading--h1 kol-heading__headline">
    Main Heading
  </h1>
  <h2 class="kol-heading__secondary-headline">
    Secondary Headline
  </h2>
</hgroup>
`;

exports[`KolHeadingFc should render with a specific level 1`] = `
<h3 class="kol-heading--h3 kol-heading__headline">
  Level 3 Heading
</h3>
`;

exports[`KolHeadingFc should render with default props 1`] = `
<h1 class="kol-heading--h1 kol-heading__headline">
  Default Heading
</h1>
`;
